### PR TITLE
fix: shopify: incorporate new shopify cart token format

### PR DESCRIPTION
**Pull request by anantjain45823**
## What are the changes introduced in this PR?
Incorporating the new format of shopify `cart_token` for identifier events only as the format change happened only for those

Write a brief explainer on your code changes.

## What is the related Linear task?

Resolves INT-2485

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [x] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by psrikanth88** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3626/test-report.html

File: src/v0/sources/shopify/transform.js [hunk 1]
```diff
@@ -208,6 +208,7 @@ const isIdentifierEvent = (event) =>
   ['rudderIdentifier', 'rudderSessionIdentifier'].includes(event?.event);
 const processIdentifierEvent = async (event, metricMetadata) => {
   if (useRedisDatabase) {
+    const cartToken = event.cartToken.split('?')[0];
```
Comment (krishna2020) [inline src/v0/sources/shopify/transform.js:211]: sanitise the input as operation is being performed on string now

Comment (utsabc) [inline src/v0/sources/shopify/transform.js:211]: Lets log `cartToken` with `writeKey`

Comment (anantjain45823) [reply src/v0/sources/shopify/transform.js]: done

Comment (anantjain45823) [reply src/v0/sources/shopify/transform.js]: done

File: src/v0/sources/shopify/transform.js [hunk 2]
```diff
@@ -208,11 +208,22 @@ const isIdentifierEvent = (event) =>
   ['rudderIdentifier', 'rudderSessionIdentifier'].includes(event?.event);
 const processIdentifierEvent = async (event, metricMetadata) => {
   if (useRedisDatabase) {
+    let { cartToken } = event;
+    if (typeof cartToken === 'string') {
+      [cartToken] = cartToken.split('?');
+    }
+    logger.debug(`{{SHOPIFY::}} writeKey: ${metricMetadata.writeKey}, cartToken: ${cartToken}`, {
```
Comment (utsabc) [inline src/v0/sources/shopify/transform.js:215]:
```suggestion
    logger.info(`{{SHOPIFY::}} writeKey: ${metricMetadata.writeKey}, cartToken: ${cartToken}`, {
```

File: src/v0/sources/shopify/util.js [hunk 3]
```diff
@@ -247,7 +249,7 @@ const checkAndUpdateCartItems = async (inputEvent, redisData, metricMetadata) =>
   const cartToken = inputEvent.token || inputEvent.id;
   if (!isDefinedAndNotNull(redisData)) {
     // eslint-disable-next-line no-param-reassign
-    redisData = await getDataFromRedis(cartToken, metricMetadata);
+    redisData = await getDataFromRedis(cartToken, metricMetadata, 'Cart Update');
```
Comment (utsabc) [inline src/v0/sources/shopify/util.js:252]: Why are we hard coding event name here can't we pass it as argument like others?

Comment (anantjain45823) [reply src/v0/sources/shopify/util.js:252]: Changed to take the shopifyTopic as parameter and pass updated name to metrics

File: src/v0/sources/shopify/transform.js (hunk 2 above)
Comment (anantjain45823) [reply src/v0/sources/shopify/transform.js]: done

**Review by utsabc — APPROVED** [review]

**Review by krishna2020 — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**